### PR TITLE
Ensure configuration is checked for Turbopack build

### PR DESCRIPTION
**Pull request by timneutkens**
## What?

Currently any configuration issue like including `.babelrc` was not highlighted during Turbopack build. This PR solves that issue as well as ensuring the warnings are not double-logged because of a change that was supposed to be development-only.

Closes NEXT-3049

### CODE REVIEW

**Comment by timneutkens** [comment]
* **#64247** <a href="https://app.graphite.dev/github/pr/vercel/next.js/64247?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ijjk** [comment]
## Failing test suites 
Commit: a832aa4271711e77f4cb1ad5615eeb45e5d4df8e

`TURBOPACK=1 pnpm test-start test/e2e/app-dir/app-routes-subrequests/app-routes-subrequests.test.ts` (turbopack)
- app-routes-subrequests > shortcuts after 5 subrequests

<details>
<summary>Expand output</summary>

  ● app-routes-subrequests › shortcuts after 5 subrequests

    next build failed with code/signal 1

      89 |           if (code || signal)
      90 |             reject(
    > 91 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      92 |             )
      93 |           else resolve()
      94 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:91:15)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/server/dev/turbopack-utils.ts [hunk 1]
```diff
@@ -64,13 +64,20 @@ export function isWellKnownError(issue: Issue): boolean {
...
 export function printNonFatalIssue(issue: Issue) {
-  // Currently only warnings will be printed, excluding if the error source
-  // is coming from foreign (node_modules) codes.
+  const nonFatalissue = formatNonFatalIssue(issue)
+  if (nonFatalissue) {
+    Log.warn(formatIssue(issue))
+  }
+}
+
+export function formatNonFatalIssue(issue: Issue): string | null {
```
Comment (sokra) [inline packages/next/src/server/dev/turbopack-utils.ts:73]: This should not to filtering and formatting together.

Instead there should be `isRelevantWarning` and return boolean.

File: packages/next/src/server/dev/hot-reloader-turbopack.ts [hunk 2]
```diff
@@ -453,6 +453,7 @@ export async function createHotReloaderTurbopack(
         manifestLoader,
         nextConfig: opts.nextConfig,
         rewrites: opts.fsChecker.rewrites,
+        logWellKnownErrors: true,
```
Comment (sokra) [inline packages/next/src/server/dev/hot-reloader-turbopack.ts:456]:
```suggestion
        logErrors: true,
```

**Review by sokra — APPROVED** [review]

**Pull request merged**